fixed most createClass warnings

vincentol opened the pull request:
### Summary

I fixed most createClass warnings that was caused from ReactJs's soon-to-be deprecated warnings. The last file that needs to be fixed is src/components/ImagePicker/index.js

Note that there are existing PropType warnings still that should also be fixed.

germanattanasio commented:
Hi @vincentol Thanks for the PR.
Can you fix the `eslint` issues?
We probably need to update the dependency version too.
You should also add your IBM email to your github profile so that the commits you have made are linked to your profile

vincentol commented:
@germanattanasio 
-- deleted unnecessary comments (I found answers to my questions) --
*Edit 3, I have fixed all the eslint issues. I have also updated every file so they no longer throw createClass and PropTypes warnings (yay for nice empty console on start). Please take a look!

jzhang300 commented on src/components/ImagePicker/index.js:
```diff
@@ -1,143 +1,64 @@
-    }
+    // if (!this.props.initialPreviewImage || this.props.initialPreviewImage === null) {
+    //   this.state;
+    // }
```
should remove comments
